sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -429,6 +434,22 @@ void release(String bundleDescriptorId, BundleProcessor bundleProcessor) {
       bundleProcessor.reset();
       cachedBundleProcessors.get(bundleDescriptorId).add(bundleProcessor);
     }
+
+    /** Shutdown all the cached {@link BundleProcessor}s, running the tearDown() functions. */
+    void shutdown() throws Exception {
+      for (ConcurrentLinkedQueue<BundleProcessor> bundleProcessors :
+          cachedBundleProcessors.values()) {
+        for (BundleProcessor bundleProcessor : bundleProcessors) {
+          // Need to reverse this since we want to call teardown in topological order.
```

Comment:
It is not necessary to reverse the list since teardown can't produce output and should not meaningfully impact other instances. Its the same with startInstance where the order doesn't matter.